src/Microsoft.Repl/Parsing/CoreParser.cs

```diff
@@ -131,5 +134,23 @@ public ICoreParseResult Parse(string commandText, int caretPosition)
 
             return new CoreParseResult(caretPosition, caretPositionWithinSelectedSection, commandText, sections, selectedSection, sectionStartLookup, quotedSections);
         }
+
+        private static void RemoveEmptyEntries(List<string> sections)
+        {
+            if (sections.Count < 2)
+            {
+                return;
+            }
+
+            // We want to remove empty spaces from the beginning, and from the middle
+            // but not from the end.
```

Comment:
Any reason to not remove it from the end?

Reply:
See my comment on the PR - https://github.com/dotnet/HttpRepl/pull/366#discussion_r463159983 - there's a lot of assumptions around it, including about how we handle suggestions (if there is no space, tab would cycle suggestions on the last keyword; if there was a space, it would be trying to handle the next keyword or option/arg).